build/ArtifactBuilder/SecurityHelpers.cs

```diff
@@ -0,0 +1,199 @@
+            }
+
+            var wtfiData = _fileInfo;
+            FileInfoPtr = Marshal.AllocCoTaskMem(Marshal.SizeOf(typeof(WinTrustFileInfo)));
+            Marshal.StructureToPtr(wtfiData, FileInfoPtr, false);
+        }
+        public void Dispose()
+        {
+            if (FileInfoPtr != IntPtr.Zero)
+            {
+                Marshal.FreeCoTaskMem(FileInfoPtr);
+                FileInfoPtr = IntPtr.Zero;
+            }
+        }
+    }
+    #endregion
+
+    enum WinVerifyTrustResult : uint
+    {
+        Success = 0,
+        ProviderUnknown = 0x800b0001,           // Trust provider is not recognized on this system
+        ActionUnknown = 0x800b0002,         // Trust provider does not support the specified action
+        SubjectFormUnknown = 0x800b0003,        // Trust provider does not support the form specified for the subject
+        SubjectNotTrusted = 0x800b0004,         // Subject failed the specified verification action
+        FileNotSigned = 0x800B0100,         // TRUST_E_NOSIGNATURE - File was not signed
+        SubjectExplicitlyDistrusted = 0x800B0111,   // Signer's certificate is in the Untrusted Publishers store
+        SignatureOrFileCorrupt = 0x80096010,    // TRUST_E_BAD_DIGEST - file was probably corrupt
+        SubjectCertExpired = 0x800B0101,        // CERT_E_EXPIRED - Signer's certificate was expired
+        SubjectCertificateRevoked = 0x800B010C,     // CERT_E_REVOKED Subject's certificate was revoked
+        UntrustedRoot = 0x800B0109          // CERT_E_UNTRUSTEDROOT - A certification chain processed correctly but terminated in a root certificate that is not trusted by the trust provider.
+    }
+
+    static class SecurityHelpers
+    {
+        private static readonly IntPtr INVALID_HANDLE_VALUE = new IntPtr(-1);
+        // GUID of the action to perform
+        private const string WINTRUST_ACTION_GENERIC_VERIFY_V2 = "{00AAC56B-CD44-11d0-8CC2-00C04FC295EE}";
+
+        [DllImport("wintrust.dll", ExactSpelling = true, SetLastError = false, CharSet = CharSet.Unicode)]
+        static extern WinVerifyTrustResult WinVerifyTrust(
+            [In] IntPtr hwnd,
+            [In][MarshalAs(UnmanagedType.LPStruct)] Guid pgActionID,
+            [In] WinTrustData pWVTData
+        );
+
+        public static bool VerifyEmbeddedSignature(string fileName)
+        {
+            WinTrustFileInfo winTrustFileInfo = null;
+            WinTrustData winTrustData = null;
+
+            try
+            {
+                // specify the WinVerifyTrust function/action that we want
+                var action = new Guid(WINTRUST_ACTION_GENERIC_VERIFY_V2);
+
+                // instantiate our WinTrustFileInfo and WinTrustData data structures
+                winTrustFileInfo = new WinTrustFileInfo(fileName);
+                winTrustData = new WinTrustData(winTrustFileInfo);
+
+                var result = WinVerifyTrust(INVALID_HANDLE_VALUE, action, winTrustData);
```

Comment:
Can we log the error code here if it fails?

Reply:
Yeah, I saw some code that takes the result and gets the corresponding system message. I'll add that back in.
Error exceptions will have a message that looks like 
```
Code signing certificate is not valid.  Error result: 800B0109 - A certificate chain processed, but terminated in a root certificate which is not trusted by the trust provider.
```

Reply:
Oh great! I would have been fine with an error code :)